### ISSUE TITLE
Support CPU Options in aws_launch_template

### ISSUE THREAD

**Issue by cread** [terraform-providers/terraform-provider-aws#6522]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add support for CPU Options to `aws_launch_template` resource.

### New or Affected Resource(s)

* aws_launch_template

### Potential Terraform Configuration

```hcl
resource "aws_launch_template" "launch" {
  name_prefix = "launch"

  cpu_options {
    core_count = 4
    thread_count = 1
  }
}
```

### References

* https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateCpuOptionsRequest.html

**Comment by praveensastry** [comment]
@bflad I have implemented this without any validations for the `core_count` and `threads_per_core` fields since the API [here](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_LaunchTemplateCpuOptionsRequest.html) doesn't specify any valid value constraints. However the docs [here](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-optimize-cpu.html?shortFooter=true#cpu-options-supported-instances-values) mentions valid values of `core_count` and `threads_per_core` per instance type. IMO there shouldn't be any validations at provider level as these values may change from time to time.

**Comment by bflad** [comment]
Support for the `cpu_options` configuration block in the `aws_launch_template` resource has been merged and will release with version 2.49.0 of the Terraform AWS Provider, tomorrow. Thanks to @praveensastry for the implementation. 👍
